sdk/core/azure-core/src/main/java/com/azure/core/http/rest/PagedFlux.java

```diff
@@ -73,12 +97,12 @@ public PagedFlux(Supplier<Mono<PagedResponse<T>>> firstPageRetriever,
      * @param mapper The mapper function to convert from type T to type S.
      * @param <S> The mapped type.
      * @return A PagedFlux of type S.
+     * @Deprecated
```

Comment:
Use the Javadoc tag `@deprecated` and follow it with text on how users should respond to this deprecation.

Reply:
We originally introduced `mapPage` for addressing storage (I think search as well) client SDK use case, afaik there was no customer ask for it. I couldn't think of a scenario where customer need to transform a PagedFlux to another. 

Recently there is a discussion around mapping error for internal use. I'm bit worried adding more and more Flux operators in PagedFlux for such decorations for internal use-cases. Though provider pattern in this PR was originally designed for state mgmt, it can also extended to solve this, it keep the bar high for SDK implementor to do the decoration but avoid leaking more operators to public surface.

Added samples for decoration use case. I'm not sure how to link those from `@deprecated` tag.